src/platforms/stm32/src/lib/gpio_driver.c

```diff
@@ -229,6 +229,25 @@ static term create_pair(Context *ctx, term term1, term term2)
     return ret;
 }
 
+static term error_tuple_maybe_gc(Context *ctx, term reason_atom)
+{
+    if (UNLIKELY(!term_is_atom(reason_atom))) {
```

Comment:
I'm still not sure which call to this function may make this condition true and execute this branch.

Reply:
There are currently no places where this could happen. This was intended as a fail-safe in case the function was misused at some point in the future by other developers, but the name change of the variable to `reason_atom` makes this clear enough. I added it a @fadushin's recommendation, but I believe with the current changes his concern is addresses:
>You might want to add a check to ensure that err-atom is always an atom, or at least document so, so that future maintainers don't pass in a tuple expecting things to work.
I have removed the check.